### PR TITLE
Make sriov-infiniband optional

### PR DESCRIPTION
**Pull request by manuelbuil**
sriov-infiniband-cni gets always deployed automatically. This PR makes it optional based on the bool env variable SRIOV_INFINIBAND_ENABLE

Signed-off-by: Manuel Buil <mbuil@suse.com>

### CODE REVIEW

**Comment by zshi-redhat** [comment]
@manuelbuil is this to make sriov-infiniband-cni be optional or sriov-infiniband?

The env variable `SRIOV_INFINIBAND_ENABLE` seems indicating SRIOV_INFINIBAND, not CNI.

There is sriovibnetwork CRD which gets deployed when installing sriov operator. If we make sriov-infiniband-cni optional, that CRD should also be optional?  And a few CRD fields in sriovnetworknodestate and sriovnetworknodepolicy are also Infiniband related, I'm not sure we can make them optional.

**Comment by manuelbuil** [comment]
> @manuelbuil is this to make sriov-infiniband-cni be optional or sriov-infiniband?
> 
> The env variable `SRIOV_INFINIBAND_ENABLE` seems indicating SRIOV_INFINIBAND, not CNI.
> 
> There is sriovibnetwork CRD which gets deployed when installing sriov operator. If we make sriov-infiniband-cni optional, that CRD should also be optional? And a few CRD fields in sriovnetworknodestate and sriovnetworknodepolicy are also Infiniband related, I'm not sure we can make them optional.

Thanks for the review. Those are good points, I'll have a look at it

File: hack/env.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ if [ -z $SKIP_VAR_SET ]; then
         export SRIOV_CNI_IMAGE=${SRIOV_CNI_IMAGE:-quay.io/openshift/origin-sriov-cni@${CNI_IMAGE_DIGEST}}
         INFINIBAND_CNI_IMAGE_DIGEST=$(skopeo inspect docker://quay.io/openshift/origin-sriov-infiniband-cni | jq --raw-output '.Digest')
         export SRIOV_INFINIBAND_CNI_IMAGE=${SRIOV_INFINIBAND_CNI_IMAGE:-quay.io/openshift/origin-sriov-infiniband-cni@${INFINIBAND_CNI_IMAGE_DIGEST}}
+        export SRIOV_INFINIBAND_CNI_ENABLE=${SRIOV_INFINIBAND_CNI_ENABLE:-true}
```
Comment (zshi-redhat) [inline hack/env.sh:11]: Perhaps we can move this line together with line 33-37 up to the top.

File: hack/env.sh [hunk 2]
```diff
@@ -21,6 +22,7 @@ if [ -z $SKIP_VAR_SET ]; then
 else
         [ -z $SRIOV_CNI_IMAGE ] && echo "SRIOV_CNI_IMAGE is empty but SKIP_VAR_SET is set" && exit 1
         [ -z $SRIOV_INFINIBAND_CNI_IMAGE ] && echo "SRIOV_INFINIBAND_CNI_IMAGE is empty but SKIP_VAR_SET is set" && exit 1
+        [ -z $SRIOV_INFINIBAND_CNI_ENABLE ] && echo "SRIOV_INFINIBAND_CNI_ENABLE is empty but SKIP_VAR_SET is set" && exit 1
```
Comment (zshi-redhat) [inline hack/env.sh:25]: I don't see why this is needed.

**Review by zshi-redhat — COMMENTED** [review]
/cc @pliurh @adrianchiris

**Comment by jjuele** [comment]
Can one of the admins verify this patch?

**Comment by SchSeba** [comment]
This is a very old PR I will close it feel free to reopen if needed

**Pull request closed without merging**